File: Nexus_Bilding_Site/src/types/product/Product.ts
import { ProductStatus } from '../shared/ProductStatus';
import { ITBISRate } from '../shared/ITBISRate';

export interface Product {
  id: string;

  name: string;
  description?: string;

  sku?: string;
  category?: string;

  price: number;
  cost?: number;

  itbisRate: ITBISRate;

  exemptionReason?: string; // obligatorio si itbisRate === 0

  stock?: number;
  lowStockThreshold?: number;

  status: ProductStatus;

  createdAt: string;
}

File: Nexus_Bilding_API/nexus_bilding_API/nexus_bilding_api.core.domain/Entities/Product.cs
using nexus_bilding_api.core.domain.Common;
using nexus_bilding_api.core.domain.Enums;

namespace nexus_bilding_api.core.domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    
    public string? Sku { get; set; }
    public string? Category { get; set; }
    
    public decimal Price { get; set; }
    public decimal? Cost { get; set; }
    
    public ItbisRate ItbisRate { get; set; }
    
    public string? ExemptionReason { get; set; }
    
    public double? Stock { get; set; }
    public double? LowStockThreshold { get; set; }
    
    public ProductStatus Status { get; set; }
    
    // Navigation Properties
    public ICollection<StockTransaction> StockTransactions { get; set; } = new List<StockTransaction>();
}
